auslib/test/web/test_json.py

```diff
@@ -0,0 +1,127 @@
+import pytest
+
+from auslib.blobs.base import createBlob
+from auslib.global_state import dbo
+from auslib.web.public.base import app
+
+
+@pytest.fixture(scope="function")
+def disable_errorhandler(monkeypatch):
+    monkeypatch.setattr(app, "error_handler_spec", {None: {}})
+
+
+@pytest.fixture(scope="module")
+def appconfig():
+    app.config["WHITELISTED_DOMAINS"] = {"good.com": ("Guardian",)}
+
+
+@pytest.fixture(scope="module")
+def guardian_db(db_schema):
+    db_schema.create_all(dbo.engine)
+    dbo.releases.t.insert().execute(
+        name="Guardian-0.5.0.0",
+        product="Guardian",
+        data_version=1,
+        data=createBlob(
+            """
+{
+    "name": "Guardian-0.5.0.0",
+    "product": "Guardian",
+    "schema_version": 10000,
+    "version": "0.5.0.0",
+    "required": true,
+    "platforms": {
+        "WINNT_x86_64": {
+            "fileUrl": "https://good.com/0.5.0.0.msi"
+        },
+        "Darwin_x86_64": {
+            "fileUrl": "https://good.com/0.5.0.0.dmg"
+        }
+    }
+}
+"""
+        ),
+    )
+    dbo.releases.t.insert().execute(
+        name="Guardian-1.0.0.0",
+        product="Guardian",
+        data_version=1,
+        data=createBlob(
+            """
+{
+    "name": "Guardian-1.0.0.0",
+    "product": "Guardian",
+    "schema_version": 10000,
+    "version": "1.0.0.0",
+    "required": true,
+    "platforms": {
+        "WINNT_x86_64": {
+            "fileUrl": "https://good.com/1.0.0.0.msi"
+        },
+        "Darwin_x86_64": {
+            "fileUrl": "https://good.com/1.0.0.0.dmg"
+        }
+    }
+}
+"""
+        ),
+    )
+    dbo.rules.t.insert().execute(
+        priority=120,
+        backgroundRate=100,
+        mapping="Guardian-0.5.0.0",
+        update_type="minor",
+        product="Guardian",
+        channel="release",
+        version="<0.5.0.0",
+        data_version=1,
+    )
+    dbo.rules.t.insert().execute(
+        priority=100, backgroundRate=100, mapping="Guardian-1.0.0.0", update_type="minor", product="Guardian", channel="release", data_version=1
```

Comment:
Nit, please use multiple lines here too. It makes it easier to compare rules.

Reply:
Hm, OK. Black actually did this. Perhaps an argument for moving to a shorter line length. In any case, I'll fix this up.